scipy/stats/_stats_py.py

```diff
@@ -1655,10 +1657,6 @@ def skewtest(a, axis=0, nan_policy='propagate', alternative='two-sided'):
 
     n = xp.asarray(_length_nonmasked(a, axis), dtype=b2.dtype, device=xp_device(a))
     n = xpx.at(n, n < 8).set(xp.nan)
-    if xp.any(xp.isnan(n)):
-        message = ("`skewtest` requires at least 8 valid observations;"
-                   "slices with fewer observations will produce NaNs.")
-        warnings.warn(message, SmallSampleWarning, stacklevel=2)
```

Comment:
Small sample warnings are emitted by the infrastructure for eager backends; no warnings are emitted for lazy backends (so we can remove restrictions on Dask arrays and JAX jit).